### ISSUE TITLE
More forgiving parser

### ISSUE THREAD

**Issue by jmuhlich** [tlambert03/ome-types#108]
Multiple times I've gotten schema validation errors from ome_types on XML that BioFormats has no trouble with. These files always turn out to be third-party-generated, of course. I suspect people aren't running the schema validation, just testing that their generated files don't blatantly error out in `showinf` or ImageJ and that those tools show the correct metadata -- I know I've done just that in the past! :flushed:  It turns out BioFormats doesn't perform schema validation when opening files and it uses a bunch of ad-hoc DOM access for parsing, so a lot of technically invalid stuff like misordered elements is accepted. Do we have any reasonable solution here? I don't want to just send people away to fix their own tools (usually the person with the file is fairly removed from the tool developer, and at any rate they have this file in hand already), but I also don't want to start littering our code with fixes for every wrinkle we encounter in the wild.

**Comment by tlambert03** [comment]
yeah, totally agree.  I think we _should_ probably make pre-validation of the XML optional (I have a local branch where I'm playing with using lxml for converting XML to a dict, which is much faster than our current `to_dict` implementation ... and just omits the validation at that step).

As for the pydantic model itself... that part is a little less clear to me, but I can imagine a number of possibilities.  If an optional field fails to validate, we could discard it, or we could have a trash bin "`invalid`" field on the root model that can just capture the data that failed to validate? ... 

I agree that we shouldn't try to _do_ the fix for them (litering our code)... and also agree that hopefully there's a slightly more forgiving middle ground that's still rather useable?

**Comment by tlambert03** [comment]
side note: one thing I'm trying to do (like in #103) is to move validation and cleaning logic off of `element_decode`, so that the pydantic model itself is smart enough to take a more generic XML->dict conversion.

**Comment by Nicholas-Schaub** [comment]
Trying to revive this a bit.

We are in the process of implementing `ome_types` into [bfio](https://github.com/polusai/bfio), and are running into issues for some types of images (such as pyramids). It would be nice to maybe make some types of OME model elements required, and everything else optional when `not strict`, and throw a warning message. Right now this is a barrier for us adopting it into our tool.

**Comment by tlambert03** [comment]
hey @Nicholas-Schaub. thanks for chiming in.  agreed, this definitely hampers partial adoption for many very reasonable use cases.  I wonder if you could share an XML or python dict or something that you're trying to load, so I could play with it locally?  (I can certainly make up something that has extra fields, etc, but might be nice to have a known use case to work with)

**Comment by Nicholas-Schaub** [comment]
I'll add it to the separate issue I opened (#125)

**Comment by Nicholas-Schaub** [comment]
Possibly resolved by #127

**Comment by tlambert03** [comment]
yeah, I'm going to close this as fixed by #127.  By using `from_xml(..., parser='lxml')`, you can now "bypass" validation of the provided xml, deferring all validation to pydantic.

discussed offline with @jmuhlich: we should eventually completely separate the concept of `validate` and `parse`.  Currently `lxml` doesn't validate and `xmlschema` does validate:
```
by default (`None`), the choices is left to the parser (which is `False` for the lxml parser)
```

but this should probably just be fully independent and not left to the parser.
